protocol/ricochet_scanner.go

```diff
@@ -19,8 +128,20 @@ func (rps *RicochetProtocolScanner) ScanProtocol(hiddenService string, osc *conf
 		report.RicochetDetected = false
 	} else {
 		osc.LogInfo("Detected possible ricochet instance\n")
-		// TODO: Actual Analysis
-		report.RicochetDetected = true
+		r := new(goricochet.Ricochet)
+		rsc := &RicochetServiceChecker{osc, r, make(chan bool)}
+		r.Init()
+		go r.ProcessMessages(rsc)
```

Comment:
Ouch good point. I checked that the connection threads would terminate, but indeed, this launches a thread launcher thread that never goes away.